modules/spark/src/main/scala/pureconfig/module/spark/spark.scala

```diff
@@ -0,0 +1,20 @@
+package pureconfig.module
+
+import scala.util.Try
+
+import org.apache.spark.sql.types.{DataType, StructType}
+
+import pureconfig.{ConfigConvert, ConfigReader, ConfigWriter}
+
+/** `ConfigConvert` instances for Spark data structures.
+  */
+package object spark {
```

Comment:
These are all `spark-sql` types. Do we expect `pureconfig-spark` to have non-sql converters too in the future?

If that's the case, I'd rather have define these in a separate `pureconfig/module/spark/sql.scala` object. While you correctly set `spark-sql` as provided, as it is now any user trying to use this module will need to provide this dependency, even if they're using things that don't depend on `spark-sql`.

Reply:
Sure I can move it down a level.

Reply:
@ruippeixotog moved 5957f793b7f9fa116277c9c29a256bf12e4350ba